### ISSUE TITLE
.hour.at() broken in some cross-timezone cases

### ISSUE THREAD

**Issue by SijmenHuizenga** [dbader/schedule#584]
```python
with mock_datetime(2023, 4, 28, 23, 15):
    # Current Berlin time: apr-28 23:15 (local) (during daylight saving)
    # Current India time: apr-29 02:45
    # Expected to run India time: apr-29 03:00
    # Next run Berlin time: apr-28 23:30
    # Expected idle seconds: 900 (15 minutes)
    schedule.clear()
    next = every().hour.at("03:00", "Asia/Kolkata").do(mock_job).next_run

    print(next)

    assert next.day == 28
    assert next.hour == 23
    assert next.minute == 30
    assert schedule.idle_seconds() == 900
```

This fails with `next = 2023-04-28 20:33:00`. Probably there is a bug somewhere, but I'm not entirely sure where it is. Does anyone know what is going on?

**Comment by dshetyo** [comment]
I am facing similar issue. See this https://github.com/dbader/schedule/issues/592

**Comment by SijmenHuizenga** [comment]
Resolved with #583 and released in 1.2.1
